flink-cyber/metron-parser-chain/parser-chains-core/src/main/java/com/cloudera/parserchains/core/ReflectiveParserBuilder.java

```diff
@@ -68,19 +70,53 @@ private void configure(Parser parser, ParserSchema parserSchema) throws InvalidP
 
             final List<ConfigValueSchema> valuesSchema = parserConfig.get(annotationKey);
 
+            validateParams(parserSchema, valuesSchema, method);
+
             if (valuesSchema != null) {
                 // execute each method with values present in schema
                 for (ConfigValueSchema value : valuesSchema) {
                     invokeMethod(parser, parserSchema, annotationKey, method, value.getValues());
                 }
+            } else if (StringUtils.hasText(key.defaultValue())){
+                final Map<String, String> value = new HashMap<>();
+                value.put(annotationKey, key.defaultValue());
+                invokeMethod(parser, parserSchema, annotationKey, method, value);
             } else if (key.required()) {
                 throw new InvalidParserException(parserSchema,
                         String.format("Required field isn't provided: %s", annotationKey));
             }
         }
     }
 
-    private void invokeMethod(Parser parser,
+    private void validateParams(ParserSchema parserSchema, List<ConfigValueSchema> valuesSchema, Method method)
+        throws InvalidParserException {
+        if (valuesSchema == null){
+          return;
+        }
+
+        final List<Parameter> paramsAnnotations = Arrays.stream(method.getParameterAnnotations())
+            .flatMap(Arrays::stream)
+            .filter(annotation -> annotation instanceof Parameter)
+            .map(annotation -> (Parameter) annotation)
+            .collect(Collectors.toList());
+
+        for (Parameter paramAnnotation : paramsAnnotations) {
+            for (ConfigValueSchema value : valuesSchema) {
+                final Map<String, String> valueMap = value.getValues();
+                final String annotationKey = paramAnnotation.key();
+
+                if (StringUtils.hasText(paramAnnotation.defaultValue()) && valueMap.get(annotationKey) == null){
+                    valueMap.put(annotationKey, paramAnnotation.defaultValue());
```

Comment:
I'm not sure it's a good idea to call a method a validation method and do assignments there.